tree/ntuple/v7/src/RField.cxx

```diff
@@ -374,6 +377,22 @@ ERNTupleUnsplitSetting GetRNTupleUnsplitSetting(TClass *cl)
    }
 }
 
+/// Used in RUnsplitField::AppendImpl() in order to record the encountered streamer info records
+class TBufferRecStreamer : public TBufferFile {
+public:
+   using RCallbackStreamerInfo = std::function<void(TVirtualStreamerInfo *)>;
+
+private:
+   RCallbackStreamerInfo fCallbackStreamerInfo;
+
+public:
+   TBufferRecStreamer(TBuffer::EMode mode, Int_t bufsiz, RCallbackStreamerInfo callbackStreamerInfo)
+      : TBufferFile(mode, bufsiz), fCallbackStreamerInfo(callbackStreamerInfo)
+   {
+   }
+   void TagStreamerInfo(TVirtualStreamerInfo *info) final { fCallbackStreamerInfo(info); }
```

Comment:
```suggestion
   void TagStreamerInfo(TVirtualStreamerInfo *info) final {
     fCallbackStreamerInfo(info);
     TBufferFile::TagStreamerInfo(info);
   }
```
would be an easy way to also record the `StreamerInfo` in the file, if (and only if) the `TBufferFile`'s parent was set to be the corresponding `TFile`.

Reply:
As discussed, it may be possible to set the backing file but it would break the layering because the fields do not know which type of page backend is used (e.g., file vs. object store).

Reply:
Then that leave the question open on how to make the files containing RNTuple to fit with the other tools (mostly `MakeProject`).

Reply:
This is true, but I plan to address this for another PR. The PR at hand should make sure that data remains readable even if dictionaries are not available.

Another PR should deal with registering streamer info of RNTuple fields in the TFile both for split and unsplit fields.